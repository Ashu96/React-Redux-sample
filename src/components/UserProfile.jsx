import React from 'react';
import PropTypes from 'prop-types';

export default function UserProfile({ profile, handleChange, handleSubmit }) {
  return (
    <div className="container">
      <h1>{profile.isFetching ? 'Loading...' : 'Submit your details here'}</h1>
      {Object.keys(profile)
        .filter(i => i !== 'isFetching')
        .map(item => (
          <input
            key={item}
            data-id={item}
            className="input"
            value={profile[item]}
            onChange={event => handleChange(event.target.value, item)}
            type="text"
            placeholder={item.toLowerCase()}
          />
        ))}
      <button type="button" className="btn" onClick={handleSubmit}>
        Save
      </button>
    </div>
  );
}

UserProfile.propTypes = {
  profile: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
    company: PropTypes.string.isRequired,
    department: PropTypes.string.isRequired,
    position: PropTypes.string.isRequired,
  }).isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};
