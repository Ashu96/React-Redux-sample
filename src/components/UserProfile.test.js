/* eslint-disable no-undef */
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import UserProfile from './UserProfile';

configure({ adapter: new Adapter() });

describe('<UserProfile />', () => {
  const props = {
    profile: {
      firstName: '',
      lastName: '',
      company: '',
      department: '',
      position: '',
      email: '',
      isFetching: true,
    },
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
  };

  it('Should match snapshot', () => {
    const tree = shallow(<UserProfile {...props} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('Should render 6 input tags', () => {
    const wrapper = shallow(<UserProfile {...props} />);
    const inputs = wrapper.find('input');
    expect(inputs.length).toBe(6);
  });

  it('Should work for submit', () => {
    const wrapper = shallow(<UserProfile {...props} />);
    const button = wrapper.find('button');
    button.simulate('click');
    expect(button.props().onClick).toHaveBeenCalledTimes(1);
  });
});
