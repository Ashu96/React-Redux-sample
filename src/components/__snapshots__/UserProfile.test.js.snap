// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> Should match snapshot 1`] = `
<div
  className="container"
>
  <h1>
    Loading...
  </h1>
  <input
    className="input"
    data-id="firstName"
    key="firstName"
    name="firstName"
    placeholder="firstname"
    type="text"
    value=""
  />
  <input
    className="input"
    data-id="lastName"
    key="lastName"
    name="lastName"
    placeholder="lastname"
    type="text"
    value=""
  />
  <input
    className="input"
    data-id="company"
    key="company"
    name="company"
    placeholder="company"
    type="text"
    value=""
  />
  <input
    className="input"
    data-id="department"
    key="department"
    name="department"
    placeholder="department"
    type="text"
    value=""
  />
  <input
    className="input"
    data-id="position"
    key="position"
    name="position"
    placeholder="position"
    type="text"
    value=""
  />
  <input
    className="input"
    data-id="email"
    key="email"
    name="email"
    placeholder="email"
    type="text"
    value=""
  />
  <button
    className="btn"
    onClick={[MockFunction]}
    type="button"
  >
    Save
  </button>
</div>
`;
